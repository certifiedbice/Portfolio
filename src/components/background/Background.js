import React,{Component} from 'react';
import Context from '../../masterProvider';
export default class Background extends Component{
static contextType=Context;
	static contextType=Context;
	componentDidMount(){
		// Start background movement.
		this.context.bgInt(true);
	}
	componentWillUnmount(){
		// End background movement.
		this.context.bgInt(false);		
	}
	render(){
		return(
			<div id='background-image-container'>
				<div id='background-image'></div>
			</div>
		);
	}
}