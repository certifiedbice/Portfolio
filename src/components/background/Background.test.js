import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Background from './Background';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<BrowserRouter><Background/></BrowserRouter>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<BrowserRouter><Background/></BrowserRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});