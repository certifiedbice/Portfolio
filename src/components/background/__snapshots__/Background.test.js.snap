// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="background-image-container"
>
  <div
    id="background-image"
  />
</div>
`;
